TheWitcher.Business/Services/HeroService.cs

```diff
@@ -229,29 +274,44 @@ public bool BuyWeapons(int heroId, int weaponsId)
             }
             return false;
         }
-        public bool SellWeapon(int heroId, int weaponsId)
+        public bool SellWeapon(int heroId, int heroWeaponsId)
         {
-            if (heroId < 0 || weaponsId < 0)
+            if (heroId < 0 || heroWeaponsId < 0)
             {
                 return false;
             }
             var hero = _heroRepository.GetItem(heroId);
-            var weapon = _weaponsRepository.GetItem(weaponsId);
-            var heroWeapon = hero.HeroWeapon.FirstOrDefault(x => x.WeaponId == weaponsId);
+            var heroWeaponObj = _heroWeaponRepository.GetItem(heroWeaponsId);
+            if (heroWeaponObj == null)
+            {
+                return false;
+            }
+            var weaponId = heroWeaponObj.WeaponId;
+            var weapon = _weaponsRepository.GetItem(weaponId.Value);
+            if (hero == null || weapon == null)
+            {
+                return false;
+            }
+            HeroWeapon heroWeapon;
+            heroWeapon = hero.HeroWeapon.FirstOrDefault(x => x.Id == heroWeaponsId);
+            if (heroWeapon == null)
+            {
+                return false;
+            }
             hero.HeroMoney += heroWeapon.PriceOfSell;
             hero.AvailableWeight += Convert.ToInt32(weapon.WeaponWeight.Value);
```

Comment:
Never convert decimal to int without exception handling.
Different number of bytes in memory, so you can get OverFlowException
You can use Truncate to remove .0 part